mountinfo/mountinfo_bsd.go

```diff
@@ -1,43 +1,39 @@
-//go:build (freebsd && cgo) || (openbsd && cgo) || (darwin && cgo)
-// +build freebsd,cgo openbsd,cgo darwin,cgo
+//go:build freebsd || openbsd || darwin
+// +build freebsd openbsd darwin
 
 package mountinfo
 
-/*
-#include <sys/param.h>
-#include <sys/ucred.h>
-#include <sys/mount.h>
-*/
-import "C"
+import "syscall"
 
-import (
-	"fmt"
-	"reflect"
-	"unsafe"
-)
+func int8SliceToString(is []int8) string {
+	var bs []byte
+	for _, i := range is {
+		if i == 0 {
+			break
+		}
+		bs = append(bs, byte(i))
+	}
+	return string(bs)
+}
 
 // parseMountTable returns information about mounted filesystems
 func parseMountTable(filter FilterFunc) ([]*Info, error) {
-	var rawEntries *C.struct_statfs
-
-	count := int(C.getmntinfo(&rawEntries, C.MNT_WAIT))
-	if count == 0 {
-		return nil, fmt.Errorf("failed to call getmntinfo")
+	const MNT_WAIT = 1
+	count, err := syscall.Getfsstat(nil, MNT_WAIT)
```

Comment:
Package `syscall` is obsoleted in favor of `x/sys/unix`.

Reply:
I changed `syscall` to `golang.org/x/sys/unix`.